pkg/util/job.go

```diff
@@ -0,0 +1,48 @@
+package util
+
+import (
+	"fmt"
+
+	seederv1alpha1 "github.com/harvester/seeder/pkg/api/v1alpha1"
+	rufio "github.com/tinkerbell/rufio/api/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// GenerateJob will generate a power action rufio job for an inventory object
+func GenerateJob(name, namespace, powerAction string) *rufio.Job {
+	var tasks []rufio.Action
+	powerOffTask := rufio.Action{
+		PowerAction: rufio.PowerHardOff.Ptr(),
+	}
+	powerOnTask := rufio.Action{
+		PowerAction: rufio.PowerOn.Ptr(),
+	}
+
+	switch powerAction {
+	case seederv1alpha1.NodePowerActionPowerOn:
+		tasks = append(tasks, powerOnTask)
+	case seederv1alpha1.NodePowerActionShutdown:
+		tasks = append(tasks, powerOffTask)
+	case seederv1alpha1.NodePowerActionReboot:
+		tasks = append(tasks, powerOffTask, powerOnTask)
+	default:
+		return nil
+	}
+
+	return &rufio.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: fmt.Sprintf("%s-%s", name, powerAction),
```

Comment:
It's better to append a hyphen here

```suggestion
			GenerateName: fmt.Sprintf("%s-%s-", name, powerAction),
```